### PR TITLE
HcalEndcapPInsert*Clusters: update sampFrac

### PR DESCRIPTION
**Pull request by sebouh137**
Changed sampling fraction for the insert from 0.98% to 2.59% in order to match the change from mixed W + Fe absorbers to all Fe absorbers (see pull request https://github.com/eic/epic/pull/744 in the epic repository)

### Briefly, what does this PR introduce?
Changes the sampling fraction for reconstructing hits in the calorimeter insert to match a change in the materials used in the detector

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [X] Other: single parameter tweak

### Please check if this PR fulfills the following:
- [X] Tests for the changes have been added
- [X] Documentation has been added / updated
- [X] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
yes.  Users must use the updated detector geometry:  https://github.com/eic/epic/pull/744
### Does this PR change default behavior?
yes.

### CODE REVIEW

File: src/detectors/FHCAL/FHCAL.cc
Comment (veprbl) [inline src/detectors/FHCAL/FHCAL.cc:116]: Needs to be updated for `HcalEndcapPInsertClusters` as well

Comment (veprbl) [reply src/detectors/FHCAL/FHCAL.cc]: Truth clusters need to be changed too.

Comment (sebouh137) [reply src/detectors/FHCAL/FHCAL.cc:116]: Hmm... It appears I was confused about what "truth clusters" means... After looking through the code, it appears to  mean that truth information is used to determine which hits are in the cluster, but it doesn't use the truth energy of the hits.  Based on this, I reintroduced the sampFrac correction in the latest commit.

**Review by veprbl — APPROVED** [review]
Moving sampling fraction correction to apply to final clusters makes sense to me. Have to trust on the values, as there are no benchmarks for this implemented.

**Comment by veprbl** [comment]
Let's wait for geometry change to be merged.

**Comment by sebouh137** [comment]
> Let's wait for geometry change to be merged.

Ok, I see that you've approved the geometry change.  Thanks for that.  As soon as the checks pass (I just merged in all of the changes from the main branch), I assume we'll be ready to merge this change as well.

**Pull request merged**
